### ISSUE TITLE
Nested variable substitution doesn't highlight correctly

### ISSUE THREAD

**Issue by stevekuznetsov** [jfcherng/Sublime-ShellScriptImproved#27]
In [`hack/common.sh`](https://github.com/openshift/origin/blob/master/hack/common.sh#L254) there is this construct:

``` sh
( os::build::internal::build_binaries "${binaries[@]+"${binaries[@]}"}" )
```

Unfortunately for us using Bash with `set -o nounset` (as most of us should be) this is the _only_ safe way to pass through an array that may or may not be empty. It seems like this is a feature, not a bug [[1]](https://www.gnu.org/software/bash/manual/html_node/Arrays.html) [[2]](http://stackoverflow.com/a/7577209). Anyway, point being this seems like it should be a common use case.

Our scopes get all mangled by it:
![ugh](https://cloud.githubusercontent.com/assets/7328370/19190150/df1f9722-8c69-11e6-90ba-84a33edc714b.png)

x-ref https://github.com/sublimehq/Packages/issues/465

**Comment by stevekuznetsov** [comment]
Weird, closing & re-opening the file gives me different behavior. Still wrong, but in a different way :)
![ok](https://cloud.githubusercontent.com/assets/7328370/19190239/68c6b550-8c6a-11e6-96d2-aed36fa5d100.png)

**Comment by jfcherng** [comment]
TL;DR

I find at least 2 mis-highlight in this issue.
![image](https://cloud.githubusercontent.com/assets/6594915/19191721/3354483a-8cd6-11e6-953d-9bc1fff2f6c4.png)

Anything I missed?

**Comment by stevekuznetsov** [comment]
Yes, it seems that multiple vars in the same line without quotes do not get highlighted correctly. Ignore yellow gutter marks, etc, they come from ShellCheck.

Good:
![better](https://cloud.githubusercontent.com/assets/7328370/19192820/89744154-8c75-11e6-8002-7322840e6872.png)

Bad:
![worse](https://cloud.githubusercontent.com/assets/7328370/19192821/8e3250aa-8c75-11e6-8d6d-4c6fe309c58e.png)

**Comment by jfcherng** [comment]
Okay. Both fixed in `1.2.11`. The dark theme has been updated.

![image](https://cloud.githubusercontent.com/assets/6594915/19194833/a4d91b74-8ce1-11e6-95ea-2b6ee5487161.png)

``` bash
${foo}/${foo}/${foo}/${foo}

function os::build::build_binaries() {
  local -a binaries=( "$@" )
  # Create a sub-shell so that we don't pollute the outer environment
  ( os::build::internal::build_binaries "${binaries[@]+"${binaries[@]}"}" )
}
```

**Comment by stevekuznetsov** [comment]
Awesome! Thanks :)
